compiler/src/dotty/tools/dotc/transform/TypeTestsCasts.scala

```diff
@@ -226,9 +230,10 @@ object TypeTestsCasts {
             def check(foundCls: Symbol): Boolean =
               if (!isCheckable(foundCls)) true
               else if (!foundCls.derivesFrom(testCls)) {
-                val unrelated = !testCls.derivesFrom(foundCls) && (
-                  testCls.is(Final) || !testCls.is(Trait) && !foundCls.is(Trait)
-                )
+                val unrelated =
+                  !testCls.derivesFrom(foundCls)
+                  && !unboxedTestCls.derivesFrom(foundCls)
+                  && (testCls.is(Final) || !testCls.is(Trait) && !foundCls.is(Trait))
```

Comment:
This line seems to change semantics of the original code:
```suggestion
                  && ((testCls.is(Final) || !testCls.is(Trait) && !foundCls.is(Trait)))
```

Reply:
I don't think so. All the suggestion does is double the pair of outer parens